core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -11046,6 +11046,14 @@ RelExpr *MergeDelete::bindNode(BindWA *bindWA)
         bindWA->setErrStatus();
         return this;
       }
+      if (CmpCommon::getDefault(COMP_BOOL_175) == DF_OFF)
```

Comment:
Does the optimizer fail to produce a plan with this change or even before? Maybe a JIRA or a comment stating the failing DELETE+INSERT merge queries will be useful. We may discussed this in the dev thread before, but is it possible that UPDATE+INSERT could have a similar issue? UPDATE+INSERT MERGE statements are now used more than before since any UPSERT statement with INDEX gets transformed to them.